src/main/java/htsjdk/samtools/util/zip/DeflaterFactory.java

```diff
@@ -43,36 +43,21 @@
  */
 public class DeflaterFactory {
 
-    private static Constructor<IntelDeflater> intelDeflaterConstructor;
+    private static final boolean isUsingIntelDeflater;
 
     static {
-        try {
-            if (Defaults.TRY_USE_INTEL_DEFLATER) {
-                final Class<IntelDeflater> clazz = (Class<IntelDeflater>) Class.forName("htsjdk.samtools.util.zip.IntelDeflater");
-                intelDeflaterConstructor = clazz.getConstructor(Integer.TYPE, Boolean.TYPE);
-            }
-        } catch (ClassNotFoundException e) {
-            intelDeflaterConstructor = null;
-        } catch (NoSuchMethodException e) {
-            intelDeflaterConstructor = null;
-        } catch (UnsatisfiedLinkError e) {
-            intelDeflaterConstructor = null;
-        }
+        isUsingIntelDeflater = Defaults.TRY_USE_INTEL_DEFLATER && (new IntelDeflater().load());
```

Comment:
Thanks for the explanation @tfenne -- I'll restore the reflection code. It may cause minor issues for downstream projects (like GATK) that use static dependency analysis via maven shade or gradle shadow to produce a slimmed-down jar for their binary distributions, since the intel classes will no longer show up as a compile-time dependency. But that sort of thing is inherently brittle anyway, and always requires special-casing to get the desired results.

The Intel deflater is optimized for compression level 1, as you probably are aware. If you're only using level 0 or max compression I can see how it would not be as useful -- though I do believe that @akiezun showed that it does outperform zlib even at the maximum compression level (@akiezun is this correct?)